dev/sparktestsupport/modules.py

```diff
@@ -359,7 +359,7 @@ def __hash__(self):
     build_profile_flags=[
         "-Pkinesis-asl",
     ],
-    environ={"ENABLE_KINESIS_TESTS": "1"},
+    environ={"ENABLE_KINESIS_TESTS": "0"},
```

Comment:
Adding tests for the `streaming-kinesis-asl` module seems to cause daily tests on branch-3.x to fail. It seems that we need to backport this patch to branch-3.x? Do you have any other better suggestions? @HyukjinKwon @dongjoon-hyun 

https://github.com/apache/spark/actions/runs/7097271047/job/19317138417#logs

![image](https://github.com/apache/spark/assets/1475305/e8d9bbfd-ea71-49cc-9090-1aba11ca3540)